src/vm/gchelpers.cpp

```diff
@@ -60,6 +60,45 @@ inline gc_alloc_context* GetThreadAllocContext()
     return & GetThread()->m_alloc_context;
 }
 
+// Checks to see if the given allocation size exceeds the
+// largest object size allowed - if it does, it throws
+// an OutOfMemoryException with a message indicating that
+// the OOM was not from memory pressure but from an object
+// being too large.
+inline void CheckObjectSize(size_t alloc_size)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+    } CONTRACTL_END;
+
+    size_t min_obj_size = sizeof(uint8_t*)   // sync block
```

Comment:
first of all, this calculation is incorrect, min_obj_size is 3-ptr size, not 4; 2ndly, this should not be calculated like this, it should be obtained from the GC side and for the places in the VM that assumes the min_obj_size, we should assert that they agree.